### PR TITLE
Deadwood Trees and lots of other goodies

### PR DESCRIPTION
**Pull request by Shnupbups**
Added Deadwood, a new type of tree that replaces the Dark Oak and Swamp trees that used to be in the Spooky Dimension.
With textures by MoonlitLamps, these trees come in 2 sizes and look fantastic!
They have everything that vanilla woods have except boats, because boats are complicated.
Other changes include:
- Added Eerie Grass, replaces vanilla grass in the spooky dimension, also exists in Tall form
- Added Brambles, which are ouchy
- Added Rotten Flesh Block
- Added Congealed Blood, a block that spawns Blood fluid when broken (similar to Ice and Water)
- Infested Tainted Stone now spawns underground in the spooky dimension
- Added more things to the non-US Englishes that were missing
- Fixed the log complaining about Witch Water Bubble Columns not having a blockstate json
- Removed the item form of Witch Water Bubble Columns because why did that exist anyway
- Ran another cleanup

### CODE REVIEW

**Review by vini2003 — APPROVED** [review]
Seems good. I didn't review the tree generation logic thoroughly but I trust you with that.

**Comment by Shnupbups** [comment]
The tree gen logic is copied straight from vanilla because I'm lazy

**Comment by valoeghese** [comment]
Can you fix the merge conflicts

File: src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+				Direction dir = Direction.Type.HORIZONTAL.random(random);
+				int g = height - random.nextInt(4);
+				int h = 2 - random.nextInt(3);
+				int startX = x;
+				int startZ = z;
+				int startY = y + height - 1;
+				
+				int i;
+				int j;
+				for (i = 0; i < height; ++i) {
```
Comment (valoeghese) [inline src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java]: change `i` to `localHeight` or something

File: src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+					if (i >= g && h > 0) {
+						startX += dir.getOffsetX();
+						startZ += dir.getOffsetZ();
+						--h;
+					}
+					
+					j = y + i;
```
Comment (valoeghese) [inline src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java]: And thus `j` is `localY`

File: src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+			} else if (!this.doesTreeFit(world, pos, height)) {
+				return false;
+			} else {
+				this.setToDirt(world, downPos);
+				this.setToDirt(world, downPos.east());
+				this.setToDirt(world, downPos.south());
+				this.setToDirt(world, downPos.south().east());
+				Direction dir = Direction.Type.HORIZONTAL.random(random);
+				int g = height - random.nextInt(4);
+				int h = 2 - random.nextInt(3);
```
Comment (valoeghese) [inline src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java]: I assume g and h are to do with leaf placement or something? Can we have more descriptive names?

File: src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+						this.addLog(posSet, world, trunkPos, bb);
+						this.addLog(posSet, world, trunkPos.east(), bb);
+						this.addLog(posSet, world, trunkPos.south(), bb);
+						this.addLog(posSet, world, trunkPos.east().south(), bb);
+					}
+				}
+				
+				for (i = -2; i <= 0; ++i) {
+					for (j = -2; j <= 0; ++j) {
+						int k = -1;
```
Comment (valoeghese) [inline src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java]: `k` is a y offset here

File: src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+					if (isAirOrLeaves(world, trunkPos)) {
+						this.addLog(posSet, world, trunkPos, bb);
+						this.addLog(posSet, world, trunkPos.east(), bb);
+						this.addLog(posSet, world, trunkPos.south(), bb);
+						this.addLog(posSet, world, trunkPos.east().south(), bb);
+					}
+				}
+				
+				for (i = -2; i <= 0; ++i) {
+					for (j = -2; j <= 0; ++j) {
```
Comment (valoeghese) [inline src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java]: `j` is a z offset here

File: src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+					BlockPos trunkPos = new BlockPos(startX, j, startZ);
+					if (isAirOrLeaves(world, trunkPos)) {
+						this.addLog(posSet, world, trunkPos, bb);
+						this.addLog(posSet, world, trunkPos.east(), bb);
+						this.addLog(posSet, world, trunkPos.south(), bb);
+						this.addLog(posSet, world, trunkPos.east().south(), bb);
+					}
+				}
+				
+				for (i = -2; i <= 0; ++i) {
```
Comment (valoeghese) [inline src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java]: `i` is an x offset here

File: src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+					this.addLeaves(world, startX, startY + 2, startZ, bb, posSet);
+					this.addLeaves(world, startX + 1, startY + 2, startZ, bb, posSet);
+					this.addLeaves(world, startX + 1, startY + 2, startZ + 1, bb, posSet);
+					this.addLeaves(world, startX, startY + 2, startZ + 1, bb, posSet);
+				}
+				
+				for (i = -3; i <= 4; ++i) {
+					for (j = -3; j <= 4; ++j) {
+						if ((i != -3 || j != -3) && (i != -3 || j != 4) && (i != 4 || j != -3) && (i != 4 || j != 4) && (Math.abs(i) < 3 || Math.abs(j) < 3)) {
+							this.addLeaves(world, startX + i, startY, startZ + j, bb, posSet);
```
Comment (valoeghese) [inline src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java]: `i` is an x offset here, `j` is a z offset here

**Review by valoeghese — CHANGES_REQUESTED** [review]
There are more things I'd like renamed but they follow the general pattern I've already done. (looking at use and thinking of a descriptive name).

**Pull request merged**